internal/pubsub/rabbitpubsub/rabbit.go

```diff
@@ -166,6 +139,54 @@ func (t *topic) SendBatch(ctx context.Context, ms []*driver.Message) error {
 	return <-errc
 }
 
+// Read from the channels established with NotifyPublish and NotifyReturn.
+// Must be called with t.mu held.
+func (t *topic) receiveFromPublishChannels(ctx context.Context, nMessages int) error {
+	// Consume all the acknowledgments for the messages we are publishing.
+	// Since SendBatch (the only caller of this method) holds the lock, we expect
+	// exactly as many acks as messages.
+	var err error
+	nAcks := 0
+	for nAcks < nMessages {
+		select {
+		case <-ctx.Done():
+			// Channel will be in a weird state (not all publish acks consumed, perhaps)
+			// so re-create it next time.
+			t.ch = nil
```

Comment:
Does `t.ch` need an explicit close?

Reply:
Good call, probably. 

Done.

Reply:
I don't see the `close`, did you push?
Can you check the file for other places where `t.ch` may need to be closed? I think there's at least one more.